File: tests/testdata/codegen/synthetic-resource-properties-pp/nodejs/synthetic-resource-properties.ts
import * as pulumi from "@pulumi/pulumi";
import * as synthetic from "@pulumi/synthetic";

const rt = new synthetic.resourceproperties.Root("rt", {});
export const trivial = rt;
export const simple = rt.res1;
export const foo = rt.res1.apply(res1 => res1.obj1?.res2?.obj2);
export const complex = rt.res1.apply(res1 => res1.obj1?.res2?.obj2?.answer);

File: tests/testdata/codegen/synthetic-resource-properties-pp/python/synthetic-resource-properties.py
import pulumi
import pulumi_synthetic as synthetic

rt = synthetic.resource_properties.Root("rt")
pulumi.export("trivial", rt)
pulumi.export("simple", rt.res1)
pulumi.export("foo", rt.res1.obj1.res2.obj2)
pulumi.export("complex", rt.res1.obj1.res2.obj2.answer)
